pytorch_lightning/utilities/apply_func.py

```diff
@@ -92,18 +93,35 @@ def move_data_to_device(batch: Any, device: torch.device):
         - :meth:`torch.Tensor.to`
         - :class:`torch.device`
     """
+
     def batch_to(data):
         # try to move torchtext data first
         if TORCHTEXT_AVAILABLE and isinstance(data, Batch):
 
             # Shallow copy because each Batch has a reference to Dataset which contains all examples
             device_data = copy(data)
             for field in data.fields:
-                # Batch contains output of Field.process(...) which is tensor hence .to(...) exists
-                device_field = getattr(data, field).to(device, non_blocking=True)
-                setattr(device_data, field, device_field)
+                # Batch contains output of Field.process(...)
```

Comment:
If someone has an idea of an easier or more "elegant" way to do this please let me know. This was the most straight forward approach that solved the issue I could come up with.

Reply:
see my suggestion